py/dml/types.py

```diff
@@ -828,16 +828,15 @@ def print_vtable_struct_declaration(self):
 class TTraitList(DMLType):
     __slots__ = ('traitname')
 
-    def __init__(self, traitname):
+    def __init__(self, traitname, const=False):
+        DMLType.__init__(self, const)
         self.traitname = traitname
 
-    const = True
```

Comment:
Always having `const` causes typed param initialization to always follow the `deep_const` route on sequences, which is completely unnecessary. We could add ad-hoc exceptions for `TTraitList`, but is there any good reason to always have `const` be true in the first place?